spec/unit/pdk/cli/validate_spec.rb

```diff
@@ -221,4 +221,32 @@
       }.to exit_zero
     end
   end
+
+  context 'with --puppet-dev' do
+    it 'activates resolved puppet version' do
+      expect {
+        PDK::CLI.run(['validate', '--puppet-dev'])
+      }.to exit_zero
+    end
```

Comment:
This test isn't actually asserting anything about activating the right puppet version yet, I assume this that part hasn't been implemented in this PR. But maybe it would be better to leave this out or mark it as pending so that we don't accidentally leave it this way later?

Reply:
Yeah, good point. I was going to address it in the next PR when I do get the puppet-version selection parts tested.